cmake/target/version/generate_version_info.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+"""
+# ===============================================================================
+# NAME: generate_version_info.py
+#
+# DESCRIPTION:  Creates a version.hpp and version.json file with version informations
+#
+# USAGE: ./generate_version_info.py /path/to/outdir/
+#
+# AUTHOR: sfregoso
+# EMAIL:  sfregoso@jpl.nasa.gov
+# DATE CREATED  : Oct. 15, 2021
+#
+# Copyright 2021, California Institute of Technology.
+# ALL RIGHTS RESERVED. U.S. Government Sponsorship acknowledged.
+# ===============================================================================
+"""
+import argparse
+import os
+import sys
+import json
+from pathlib import Path
+
+from fprime_ac.utils.version import (
+    FALLBACK_VERSION,
+    get_fprime_version,
+    get_project_version,
+    get_library_versions,
+)
+
+
+def create_version_file_hpp(output_dir, framework_version, project_version):
+    """
+    Create the version file using the provided name and path.
+    """
+    version_hpp = Path(output_dir) / "version.hpp.tmp"
+    # Open file for writing
+    with open(version_hpp, "w") as fid:
+        fid.write("/*\n")
+        fid.write(
+            f"    This file has been autogenerated using [{os.path.basename(__file__)}].\n"
+        )
+        fid.write("    This file may be overwritten.\n")
+        fid.write("*/\n")
+        fid.write("#ifndef _VERSION_HPP_\n")
+        fid.write("#define _VERSION_HPP_\n")
+        fid.write("\n")
+        fid.write(f'static const char* FRAMEWORK_VERSION = "{framework_version}";\n')
+        fid.write(f'static const char* PROJECT_VERSION = "{project_version}";\n')
+        lib_versions = get_library_versions()
+        # TODO: LIBRARY_VERSIONS is unused for now - remove the unused attribute when used in Version component
+        fid.write("static const char* LIBRARY_VERSIONS[] __attribute__((unused)) = {")
```

Comment:
`__attribute__((unused))` is kind of icky, but I don't know what of a better option. This will need to be removed once the Version component uses it.

Reply:
Done